app/uk/gov/hmrc/economiccrimelevyaccount/controllers/AccountController.scala

```diff
@@ -38,66 +39,59 @@ class AccountController @Inject() (
-            case None    => None
-          }
-
-          financialDataService.retrieveFinancialData.map {
-            case None               =>
-              auditAccountViewed(obligationData, None)
-              Ok(
-                view(
-                  request.eclRegistrationReference,
-                  ViewUtils.formatLocalDate(registrationDate),
-                  latestObligationData,
-                  None
-                )
-              )
-            case Some(dataResponse) =>
-              auditAccountViewed(obligationData, Some(dataResponse))
-              Ok(
-                view(
-                  request.eclRegistrationReference,
-                  ViewUtils.formatLocalDate(registrationDate),
-                  latestObligationData,
-                  dataResponse.documentDetails match {
-                    case Some(_) => financialDataService.getLatestFinancialObligation(dataResponse)
-                    case None    => None
-                  }
-                )
-              )
-          }
-        }
-    }
+    implicit val hc: HeaderCarrier = CorrelationIdHelper.getOrCreateCorrelationID(request)
+    (for {
+      registrationDate             <-
+        enrolmentStoreProxyService.getEclRegistrationDate(request.eclReference).asResponseError
+      obligationDataOption         <- eclAccountService.retrieveObligationData.asResponseError
+      latestObligationDetailsOption = obligationDataOption.flatMap(getLatestObligation)
+      financialDataOption          <- eclAccountService.retrieveFinancialData.asResponseError
+      response                      =
+        determineResponse(registrationDate, latestObligationDetailsOption, financialDataOption)
+      _                             =
+        auditService
+          .auditAccountViewed(request.internalId, request.eclReference, obligationDataOption, financialDataOption)
+    } yield response).fold(
+      presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
+      result => result
+    )
   }
 
-  private def auditAccountViewed(obligationData: Option[ObligationData], financialData: Option[FinancialDataResponse])(
-    implicit request: AuthorisedRequest[_]
-  ): Unit =
-    auditConnector.sendExtendedEvent(
-      AccountViewedAuditEvent(
-        internalId = request.internalId,
-        eclReference = request.eclRegistrationReference,
-        obligationDetails = obligationData.map(_.obligations.flatMap(_.obligationDetails)).toSeq.flatten,
-        financialDetails = financialData match {
-          case Some(details) => mapAccountViewedAuditFinancialDetails(details)
-          case None          => None
-        }
-      ).extendedDataEvent
-    )
+  private def determineResponse(
+    registrationDate: LocalDate,
+    latestObligationDetailsOption: Option[ObligationDetails],
+    financialDataOption: Option[FinancialData]
+  )(implicit request: AuthorisedRequest[_]): Result =
+    financialDataOption
```

Comment:
You could get away with a single statement here instead of a getOrElse by using a match